redis_commands.c

```diff
@@ -1818,141 +1818,82 @@ int redis_key_str_arr_cmd(INTERNAL_FUNCTION_PARAMETERS, RedisSock *redis_sock,
  * timeout value.  This can handle various SUNION/SUNIONSTORE/BRPOP type
  * commands. */
 static int gen_varkey_cmd(INTERNAL_FUNCTION_PARAMETERS, RedisSock *redis_sock,
-                          char *kw, int kw_len, int min_argc, int has_timeout,
+                          char *kw, int kw_len, int min_argc, zend_bool has_timeout,
                           char **cmd, int *cmd_len, short *slot)
 {
-    zval *z_args, *z_ele, ztimeout = {0};
-    HashTable *ht_arr;
-    char *key;
-    int key_free, i, tail;
-    size_t key_len;
-    int single_array = 0, argc = ZEND_NUM_ARGS();
+    zval *argv = NULL, ztimeout = {0}, *zv;
     smart_string cmdstr = {0};
     short kslot = -1;
-    zend_string *zstr;
+    int single_array;
+    int argc = 0;
 
-    if (argc < min_argc) {
-        zend_wrong_param_count();
-        return FAILURE;
-    }
+    ZEND_PARSE_PARAMETERS_START(min_argc, -1)
+        Z_PARAM_VARIADIC('*', argv, argc)
+    ZEND_PARSE_PARAMETERS_END_EX(return FAILURE);
 
-    // Allocate args
-    z_args = emalloc(argc * sizeof(zval));
-    if (zend_get_parameters_array(ht, argc, z_args) == FAILURE) {
-        efree(z_args);
-        return FAILURE;
-    }
+    single_array = argc == 1 + !!has_timeout && Z_TYPE(argv[0]) == IS_ARRAY;
```

Comment:
I can't parse this line 😄 Could you add parenthesis?

Reply:
It made sense to me at the time.  Pushed updated logic :joy: